### ISSUE TITLE
ma.where type handling incorrect (migrated from Trac #1207)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2760]
Original ticket http://projects.scipy.org/numpy/ticket/1207
Reported 2009-08-22 by atmention:rgommers, assigned to atmention:pierregm.

When both `x` and `y` arguments are not masked arrays, the type is determined by `np.max(type1, type2). This does not follow the NumPy casting rules, the solution is to replace`np.max`by`np.find_common_type``.

Example:

```
>>> x = np.arange(4, dtype=np.int32)
>>> y = np.arange(4, dtype=np.float32) * 2.2
>>> np.ma.where(x > 1.5, y, x).dtype
dtype('float32')
>>> np.find_common_type([np.int32, np.float32], [])
dtype('float64')
```

**Comment by thouis** [comment]
Comment in Trac by atmention:pierregm, 2009-08-30

Good call. Fixed in #7350

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-05-31
